modules/ROOT/pages/http/http-authentication.adoc

```diff
@@ -2,357 +2,496 @@
 
-In the following procedure, you configure a number of options, including these:
+In the following procedure, you can configure a number of options, including:
 
-*  *localauthorizationUrl*
+*  *Local authorization URL*
 +
 Defines a URL in your application that listens for incoming requests.
 +
 * *Authorization URL*
 +
-https://developer.github.com/v3/oauth/#web-application-flow[Provided by Github], this URL redirects the user request from the Mule client app to the Authorization URL of the Github authentication server.
+https://developer.github.com/v3/oauth/#web-application-flow[Provided by GitHub], this URL redirects the user request from the Mule client app to the Authorization URL of the GitHub authentication server.
 +
 * *Token URL*
 +
 The Mule client app sends the token to the *Token URL* that you configure in the Mule client app.
 
-*To configure the Mule client app for accessing the Github authentication server:*
+The following sections help you configure the Mule client app for accessing the GitHub authentication server.
 
-[.ex]
-=====
-[discrete.view]
-=== Studio Visual Editor
+=== OAuth 2 Authorization - Studio Visual Editor
 
-. Add the OAuth Module from the palette.
-. In Studio, select the HTTP Request Configuration global element where you want to use the OAuth authorization code grant type.
-. On the *Authentication* combo box, select `Authorization code grant type`.
-. Set the following options:
+. Drag an *HTTP* > *Request* operation from the Mule Palette to the Process area of the Studio flow. This operation is known as the HTTP Request connector.
+. Expand the Package Explorer and double-click the `pom.xml` file.
+. At the end of the `<dependencies>` section and before the `</dependencies>` statement,
+add these statements to enable OAuth options for the HTTP Request connector:
 +
-* Client Id: Type the client Id that Github provided when you registered the app.
-* Client Secret: Type the client secret that Github provided when you registered the app.
-* Local callback URL: `+http://localhost:8082/callback+`
+[[setpomfile]]
+[source,xml,linenums]
+----
+<dependency>
+    <groupId>org.mule.modules</groupId>
+    <artifactId>mule-oauth-module</artifactId>
+    <version>1.1.2</version>
+    <classifier>mule-plugin</classifier>
+</dependency>
+----
++
+. In the properties editor for *Connector Configuration*, click the green plus icon (or click the Edit icon if you have already created a configuration).
+. Set the *Authentication* field to *Authorization code grant type*.
+. Set the following required fields:
++
+** *External callback url*: `+http://localhost:8082/callback+`
```

Comment:
`localhost` is not a good example for an external url. Change it to `myapp.mycompany.com` or something along those lines

Reply:
Fixed globally